File: app/src/main/java/com/wenitech/cashdaily/ui/Adapter/RecyclerViewGastoAdapter.java
package com.wenitech.cashdaily.ui.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.wenitech.cashdaily.Data.model.MovimientoCaja;
import com.wenitech.cashdaily.R;
import com.wenitech.cashdaily.databinding.ItemMovimientoCajaBinding;

import java.text.NumberFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public class RecyclerViewGastoAdapter extends FirestoreRecyclerAdapter<MovimientoCaja, RecyclerViewGastoAdapter.mViewHolder> {

    public RecyclerViewGastoAdapter(@NonNull FirestoreRecyclerOptions<MovimientoCaja> options) {
        super(options);
    }

    @Override
    protected void onBindViewHolder(@NonNull final mViewHolder holder, int position, @NonNull MovimientoCaja model) {

        String stringValor;
        String stringDate;

        NumberFormat numberFormatMoney = NumberFormat.getCurrencyInstance();
        stringValor = numberFormatMoney.format(model.getValor());

        SimpleDateFormat dateFormat = new SimpleDateFormat("yyy-MM-dd");
        Date date = model.getFecha().toDate();
        stringDate = dateFormat.format(date);

        holder.binding.tvValor.setText(stringValor);
        holder.binding.tvDate.setText(stringDate);
        // On Click Listener
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
    }

    @NonNull
    @Override
    public mViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemMovimientoCajaBinding viewBinding = ItemMovimientoCajaBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new mViewHolder(viewBinding);
    }

    public class mViewHolder extends RecyclerView.ViewHolder{
        private ItemMovimientoCajaBinding binding;
        public mViewHolder(@NonNull ItemMovimientoCajaBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/wenitech/cashdaily/features/caja/ui/adapter/RecyclerViewGastoAdapter.kt
package com.wenitech.cashdaily.features.caja.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.firebase.ui.firestore.FirestoreRecyclerAdapter
import com.wenitech.cashdaily.R
import com.wenitech.cashdaily.base.BaseViewHolder
import com.wenitech.cashdaily.data.model.MovimientoCaja
import com.wenitech.cashdaily.databinding.ItemMovimientoCajaBinding
import java.text.NumberFormat
import java.text.SimpleDateFormat

class RecyclerViewGastoAdapter(private val listener: Listener) : RecyclerView.Adapter<BaseViewHolder<*>>() {

    interface Listener {
        fun onClickItemMovementBox(item: MovimientoCaja, position: Int)
    }

    private var data: List<MovimientoCaja> = listOf()

    fun setData(data: List<MovimientoCaja>) {
        this.data = data
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<*> {
        val viewBinding = ItemMovimientoCajaBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return mViewHolder(viewBinding)
    }

    override fun onBindViewHolder(holder: BaseViewHolder<*>, position: Int) {
        when (holder) {
            is mViewHolder -> holder.bind(data[position], position)
        }
    }

    private inner class mViewHolder(val binding: ItemMovimientoCajaBinding) : BaseViewHolder<MovimientoCaja>(binding.root) {

        override fun bind(item: MovimientoCaja, position: Int) {


            val stringValor: String
            val stringDate: String
            val numberFormatMoney = NumberFormat.getCurrencyInstance()
            stringValor = numberFormatMoney.format(item.valor)
            val dateFormat = SimpleDateFormat("dd MMMM, yyyy")
            val date = item.fecha!!.toDate()
            stringDate = dateFormat.format(date)

            binding.tvValor.text = stringValor
            binding.tvDate.text = stringDate
            binding.tvDescription.text = item.descripcion
            if (item.agregarDinero!!) {
                binding.imageViewPerfil2.setImageResource(R.drawable.ic_asset_trending_up_verde)
            } else {
                binding.imageViewPerfil2.setImageResource(R.drawable.ic_asset_trending_down_rojo)
            }

            // On Click Listener
            binding.root.setOnClickListener {
                listener.onClickItemMovementBox(item, position)
            }


        }
    }

    override fun getItemCount(): Int {
        return data.size
    }


}